docs/apm-data-security.asciidoc

```diff
@@ -455,4 +483,97 @@ Please see the relevant agent's documentation for more information and examples:
 * Node.js: {apm-node-ref-v}/agent-api.html#apm-add-filter[`addFilter()`].
 // * PHP: {apm-php-ref-v}[]
 * Python: {apm-py-ref-v}/sanitizing-data.html[custom processors].
-// * Ruby: {apm-ruby-ref-v}/[]
+* Ruby: {apm-ruby-ref-v}/api.html#api-agent-add-filter[`add_filter()`].
+
+// ****************************************************************
+
+[[data-security-delete]]
+==== Delete sensitive data
+
+If you accidentally ingest sensitive data, follow these steps to remove or redact the offending data:
+
+. Stop collecting the sensitive data.
+Use the *remedy* column of the <<sensitive-fields,sensitive fields>> table to determine how to stop collecting
+the offending data.
+
+. Delete or redact the ingested data. With data collection fixed, you can now delete or redact the offending data:
++
+* <<redact-field-data>>
+* <<delete-doc-data>>
+
+[float]
+[[redact-field-data]]
+===== Redact specific fields
+
+To redact sensitive data in a specific field, use the {ref}/docs-update-by-query.html[update by query API].
+
+For example, the following query removes the `client.ip` address
+from APM documents in the `logs-apm.error-default` data stream:
+
+[source, console]
+----
+POST /logs-apm.error-default/_update_by_query
+{
+  "query": {
+    "exists": {
+      "field": "client.ip"
+    }
+  }
+  "script": {
+    "source": "ctx._source.client.ip = params.redacted",
+    "params": {
+      "redacted": "[redacted]"
+    }
+  }
+}
+----
+
+Or, perhaps you only want to redact IP addresses from European users:
+
+[source, console]
+----
+POST /logs-apm.error-default/_update_by_query
+{
+  "query": {
+    "term": {
+      "client.geo.continent_name": {
+        "value": "Europe"
+      }
+    }
+  },
+  "script": {
+    "source": "ctx._source.client.ip = params.redacted",
+    "params": {
+      "redacted": "[redacted]"
+    }
+  }
+}
+----
+
+See {ref}/docs-update-by-query.html[update by query API] for more information and examples.
+
+[float]
+[[delete-doc-data]]
+===== Delete {es} documents
```

Comment:
Do you see a use case where someone would want to delete entire documents?

Reply:
Nothing immediately comes to my mind, but I don't see harm in having the examples.

Reply:
👍 Cool beans